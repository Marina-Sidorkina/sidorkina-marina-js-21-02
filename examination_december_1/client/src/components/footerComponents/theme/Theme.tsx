import React, { useContext } from 'react';
import { Switch } from 'antd';
import { useTranslation } from 'react-i18next';
import styles from './Theme.module.scss';
import { ThemeContext } from '../../../contexts/ThemeContext';
import '../../../locale/i18next';

const Theme = () => {
  const themeContext = useContext(ThemeContext);
  const { t } = useTranslation();

  return (
    <div className={styles.theme}>
      <span className={styles.text}>
        { t('theme', {}) }
      </span>
      <Switch
        className="theme__toggle"
        onChange={(switched: boolean) => (
          themeContext.toggleTheme && themeContext.toggleTheme(switched)
        )}
        checked={themeContext.darkTheme}
      />
    </div>
  );
};

export default Theme;
