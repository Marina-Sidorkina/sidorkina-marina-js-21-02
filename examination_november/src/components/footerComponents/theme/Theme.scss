.theme {
  &__text {
    margin-right: 10px;
  }
}
