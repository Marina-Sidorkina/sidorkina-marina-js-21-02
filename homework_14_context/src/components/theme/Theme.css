.theme {
  display: flex;
  align-items: center;
}

.theme_dark {
  color: white;
}

.theme__label {
  margin-right: 8px;
  font-size: 24px;
}

.theme__input {
  transform: scale(1.7);
  margin-right: 7px;
}
