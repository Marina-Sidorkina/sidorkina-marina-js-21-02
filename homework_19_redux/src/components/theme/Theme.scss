.theme {
  display: flex;
  align-items: center;

  &_dark {
    color: white;
  }

  &__label {
    margin-right: 8px;
    font-size: 24px;
  }

  &__input {
    transform: scale(1.7);
    margin-right: 7px;
  }
}
