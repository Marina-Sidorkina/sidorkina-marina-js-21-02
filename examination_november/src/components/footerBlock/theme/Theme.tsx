import React from 'react';
import './Theme.scss';
import { Switch } from 'antd';

const Theme = () => (
  <div className="theme">
    <span className="theme__text">Темная тема</span>
    <Switch
      className="theme__toggle"
      onChange={(switched: boolean) => { console.log(switched); }}
    />
  </div>
);

export default Theme;
