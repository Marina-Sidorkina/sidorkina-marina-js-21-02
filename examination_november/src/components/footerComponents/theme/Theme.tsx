import React, { useContext } from 'react';
import './Theme.scss';
import { Switch } from 'antd';
import { ThemeContext } from '../../../contexts/ThemeContext';

const Theme = () => {
  const themeContext = useContext(ThemeContext);

  return (
    <div className="theme">
      <span className="theme__text">Тёмная тема</span>
      <Switch
        className="theme__toggle"
        onChange={(switched: boolean) => (
          themeContext.toggleTheme && themeContext.toggleTheme(switched)
        )}
        checked={themeContext.darkTheme}
      />
    </div>
  );
};

export default Theme;
