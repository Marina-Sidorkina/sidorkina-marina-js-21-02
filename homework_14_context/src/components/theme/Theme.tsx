import React from "react";
import "./Theme.css";

class Theme extends React.Component {
  render() {
    return (
      <form className="theme">
        <label className="theme__label" htmlFor="theme">Тёмная тема</label>
        <input className="theme__input" type="checkbox" id="theme"/>
      </form>
    );
  }
}

export default Theme;
