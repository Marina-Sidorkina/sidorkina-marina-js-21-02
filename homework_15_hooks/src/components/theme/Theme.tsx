import React, {useContext} from "react";
import "./Theme.scss";
import { ThemeContext } from "../../contexts/ThemeContext";

const Theme = () => {
  const themeContext = useContext(ThemeContext);

  return (
    <form className={ `theme ${ themeContext.darkTheme ? "theme_dark" : "" }`}>
      <label className="theme__label" htmlFor="theme">Тёмная тема</label>
      <input className="theme__input"
             type="checkbox"
             id="theme"
             checked={ themeContext.darkTheme }
             onChange={ (evt) =>
               themeContext.toggleTheme && themeContext.toggleTheme(evt.target.checked) }/>
    </form>
  );
}

export default Theme;
