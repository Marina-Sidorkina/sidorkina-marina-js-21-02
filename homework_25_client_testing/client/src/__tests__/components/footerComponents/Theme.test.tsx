import React from 'react';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { shallow, configure } from 'enzyme';
import Theme from '../../../components/footerComponents/theme/Theme';

configure({ adapter: new Adapter() });

describe('Theme component test', () => {
  test('should render Theme', () => {
    const wrapper = shallow(<Theme />);
    expect(wrapper.length).toEqual(1);
  });

  test('should render Theme', () => {
    const wrapper = shallow(<Theme />);
    expect(wrapper).toMatchSnapshot();
  });
});
