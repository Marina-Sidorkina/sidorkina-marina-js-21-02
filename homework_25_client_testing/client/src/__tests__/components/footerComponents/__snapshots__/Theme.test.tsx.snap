// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Theme component test should render Theme 1`] = `
<div
  className="theme"
>
  <span
    className="text"
  >
    Dark Theme
  </span>
  <Switch
    onChange={[Function]}
  />
</div>
`;
